arch/arm/soc/st_stm32/stm32f0/soc.c

```diff
@@ -41,9 +41,12 @@ void relocate_vector_table(void)
 #ifndef CONFIG_IS_BOOTLOADER
 	extern char _ram_vector_start[];
 
-	size_t vector_size = (size_t)_vector_end - (size_t)_vector_start;
+	size_t vector_size = (size_t)_vector_end - (size_t)_vector_start -
```

Comment:
The size of the vector table should not change if I add a start offset. Why is the offset subtracted from the original size?

Reply:
The vector table does not start at vector_start, but at vector_start+CONFIG_TEXT_SECTION_OFFSET, so the length of the copy must also be recalculated

Reply:
See:
https://github.com/zephyrproject-rtos/zephyr/blob/master/include/arch/arm/cortex_m/scripts/linker.ld#L97

Reply:
I would suggest:

. = CONFIG_TEXT_SECTION_OFFSET;
_vector_start = .;

which makes the _vector_start label the same as CONFIG_TEXT_SECTION_OFFSET.

Reply:
This is another story... indeed looking at eb48a0a73c11c2a9cd4c3c91864ca4e0cf52dae8 we can simply fix this like this.